core/src/test/scala/unit/kafka/coordinator/group/PartitionWriterImplTest.scala

```diff
@@ -0,0 +1,225 @@
+      ArgumentMatchers.eq(AppendOrigin.COORDINATOR),
+      recordsCapture.capture(),
+      ArgumentMatchers.eq(1),
+      ArgumentMatchers.eq(RequestLocal.NoCaching)
+    )).thenReturn(Map(tp -> LogAppendResult(new LogAppendInfo(
+      Optional.empty(),
+      10,
+      OptionalInt.empty(),
+      RecordBatch.NO_TIMESTAMP,
+      -1L,
+      RecordBatch.NO_TIMESTAMP,
+      -1L,
+      RecordConversionStats.EMPTY,
+      CompressionType.NONE,
+      CompressionType.NONE,
+      -1,
+      -1,
+      false,
+      -1L,
+      Collections.emptyList(),
+      "",
+      LeaderHwChange.INCREASED
+    ))))
+
+    val records = List(
+      ("k0", "v0"),
+      ("k1", "v1"),
+      ("k2", "v2"),
+    )
+
+    assertEquals(11, partitionRecordWriter.append(
+      tp,
+      records.asJava
+    ))
+
+    verify(replicaManager).maybeCompletePurgatories(
+      tp,
+      LeaderHwChange.INCREASED
+    )
+
+    val batch = recordsCapture.getValue.getOrElse(tp,
+      throw new AssertionError(s"No records for $tp"))
+    assertEquals(1, batch.batches().asScala.toList.size)
+
+    val receivedRecords = batch.records.asScala.map { record =>
+      (
+        Charset.defaultCharset().decode(record.key).toString,
+        Charset.defaultCharset().decode(record.value).toString,
+      )
+    }.toList
+
+    assertEquals(records, receivedRecords)
+  }
+
+  @Test
+  def testWriteRecordTooLarge(): Unit = {
+    val tp = new TopicPartition("foo", 0)
+    val replicaManager = mock(classOf[ReplicaManager])
+    val partitionRecordWriter = new PartitionWriterImpl(
+      replicaManager,
+      new StringKeyValueSerializer(),
+      CompressionType.NONE,
+      Time.SYSTEM
+    )
+
+    val maxBatchSize = 16384
+    when(replicaManager.getLogConfig(tp)).thenReturn(Some(LogConfig.fromProps(
+      Map(TopicConfig.MAX_MESSAGE_BYTES_CONFIG -> maxBatchSize).asJava,
+      new Properties()
+    )))
+
+    val randomBytes = TestUtils.randomBytes(maxBatchSize + 1)
+    // We need more than one record here because the first record
+    // is always allowed by the MemoryRecordsBuilder.
+    val records = List(
+      ("k0", new String(randomBytes)),
+      ("k1", new String(randomBytes)),
+    )
+
+    assertThrows(classOf[RecordTooLargeException],
+      () => partitionRecordWriter.append(tp, records.asJava))
+  }
+
+  @Test
+  def testWriteEmptyRecordList(): Unit = {
+    val tp = new TopicPartition("foo", 0)
+    val replicaManager = mock(classOf[ReplicaManager])
+    val partitionRecordWriter = new PartitionWriterImpl(
+      replicaManager,
+      new StringKeyValueSerializer(),
+      CompressionType.NONE,
+      Time.SYSTEM
+    )
+
+    when(replicaManager.getLogConfig(tp)).thenReturn(Some(LogConfig.fromProps(
+      Collections.emptyMap(),
+      new Properties()
+    )))
+
+    assertThrows(classOf[IllegalStateException],
+      () => partitionRecordWriter.append(tp, List.empty.asJava))
+  }
+
+  @Test
+  def testInexistentPartition(): Unit = {
```

Comment:
nit: Nonexistent

Reply:
Also this is testing the log config doesn't exist right? Is this only the case if the partition is also nonexistent?

Reply:
Right but the config does not exist because the partition does not exist either.